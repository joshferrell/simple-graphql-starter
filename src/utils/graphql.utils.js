
const getSchemaValues = (accumulator, { name, type, description }) => {
    accumulator[name] = {
        type,
        description
    };
    return accumulator;
};

export const getInputFields = fields =>
    fields
        .filter(field => field.isInput)
        .reduce((accumulator, field) => getSchemaValues(accumulator, field), {});

export const getOutputFields = fields =>
    fields
        .filter(field => field.isOutput)
        .reduce((accumulator, field) => getSchemaValues(accumulator, field), {});
