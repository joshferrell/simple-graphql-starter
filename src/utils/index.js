export {
    hashPassword,
    generateJwt,
    decodeJwt,
    comparePassword
} from './auth.utils';

export {
    getInputFields,
    getOutputFields
} from './graphql.utils';
