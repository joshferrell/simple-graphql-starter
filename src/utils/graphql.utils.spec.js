import { getInputFields, getOutputFields } from './graphql.utils';

const testSchema = [
    {
        name: 'id',
        type: 'id',
        description: 'The UUID of the account',
        isInput: false,
        isOutput: true
    },
    {
        name: 'email',
        type: 'string',
        description: 'The email of the account, must be unique',
        isInput: true,
        isOutput: true
    },
    {
        name: 'password',
        type: 'string',
        description: 'Login password of the account',
        isInput: true,
        isOutput: false
    }
];

describe('graph-ql utilities', () => {
    it('should filter out input fields', () =>
        expect(getInputFields(testSchema)).toMatchSnapshot()
    );

    it('should filter out output fields', () =>
        expect(getOutputFields(testSchema)).toMatchSnapshot()
    );

    it('should only return required graphql values', () => {
        const output = getOutputFields(testSchema);
        return expect(output).toMatchObject({
            id: {
                type: 'id',
                description: 'The UUID of the account'
            },
            email: {
                type: 'string',
                description: 'The email of the account, must be unique'
            }
        });
    });
});
