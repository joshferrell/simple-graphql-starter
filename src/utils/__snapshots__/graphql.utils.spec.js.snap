// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graph-ql utilities should add non null types to required schema 1`] = `
Object {
  "id": Object {
    "description": "The UUID of the account",
    "type": "ID!",
  },
}
`;

exports[`graph-ql utilities should filter out input fields 1`] = `
Object {
  "email": Object {
    "description": "The email of the account, must be unique",
    "type": "string",
  },
  "password": Object {
    "description": "Login password of the account",
    "type": "string",
  },
}
`;

exports[`graph-ql utilities should filter out output fields 1`] = `
Object {
  "email": Object {
    "description": "The email of the account, must be unique",
    "type": "string",
  },
  "id": Object {
    "description": "The UUID of the account",
    "type": "id",
  },
}
`;
